CognitiveVRTest/Plugins/CognitiveVR/Source/CognitiveVR/Private/network/network.cpp

```diff
@@ -110,20 +143,45 @@ void Network::OnCallReceivedAsync(FHttpRequestPtr Request, FHttpResponsePtr Resp
 					localCacheRequest->SetHeader("Content-Type", TEXT("application/json"));
 					localCacheRequest->SetHeader("Authorization", AuthValue);
 
-					localCacheRequest->OnProcessRequestComplete().BindRaw(this, &Network::OnLocalCacheCallReceivedAsync);
-					localCacheRequest->ProcessRequest();
+					
+					world->GetTimerManager().SetTimer(TimerHandleShortDelay, FTimerDelegate::CreateRaw(this, &Network::RequestLocalCache), LocalCacheLoopDelay, false);
+					
+					VariableDelayMultiplier = 0;
 				}
 			}
+			else
+			{
+			}
 		}
-		else
+		//when we get here, we start the variable delay, 60 seconds each time, and just let the rest of the block happen
+		else 
 		{
-			if (responseCode < 500) {return;}
+			if (responseCode < 500 && responseCode != 0) {return;}
 
 			if (Request == nullptr) { return; }
 
 			//hold a pointer to a request specifically for uploading cached data
 			//if request exist, cancel it
+			auto world = ACognitiveVRActor::GetCognitiveSessionWorld();
+			world->GetTimerManager().ClearTimer(TimerHandleShortDelay);
+			if (!IsServerUnreachable)
+			{
+				IsServerUnreachable = true;
+				++VariableDelayMultiplier;
+
+				
+				if (world != nullptr && !world->GetTimerManager().IsTimerActive(TimerHandle))
+				{
+					float VariableDelay = 60 * VariableDelayMultiplier;
```

Comment:
My suggestion is to move the `60` to a well-named constant somewhere for for readability and for centralization of future changes.

Reply:
good call